File: fcfs.c
/*
  FCFS Disk Scheduling Implementation

*/

#include <stdio.h>
#include <stdlib.h>


int main(){
  int queue[100], q_size, head, seek =0, diff;
  float avg;

  printf("%s\n", "***FCFS Disk Scheduling Algorithm***");

  printf("%s\n", "Enter the size of the queue");
  scanf("%d", &q_size);

  printf("%s\n", "Enter queue elements");
  for(int i=1; i<=q_size; i++){
    scanf("%d",&queue[i]);
  }

  printf("%s\n","Enter initial head position");
  scanf("%d", &head);

  queue[0]=head;

  for(int j=0; j<=q_size-1; j++){
    diff = abs(queue[j]-queue[j+1]);
    seek += diff;
    printf("Move from %d to %d with Seek %d\n",queue[j],queue[j+1],diff);
  }

  printf("\nTotal seek time is %d\t",seek);
  avg = seek/(float)q_size;
  printf("\nAverage seek time is %f\t", avg);

  return 0;
}

File: disk_sch_java/src/fcfs.java
import java.util.Scanner;

public class fcfs {

    public static void main(String [] args){
        Scanner input = new Scanner(System.in);
        int seek = 0;
        int diff = 0;
        int head = 0;
        float avg;
        int[] queue;

        System.out.println("FCFS Disk Scheduling");

        System.out.println("Enter queue size");
        int qSize = Integer.parseInt(input.nextLine());

        queue = new int[qSize];

        System.out.println("Input Queue elements");
        for(int i=0; i<qSize; i++){
            queue[i] = Integer.parseInt(input.nextLine());
        }

        System.out.println("Enter initial head position");
        head  = Integer.parseInt(input.nextLine());

        queue[0] = head;

        for(int j=0; j<qSize-1; j++){
            diff = Math.abs(queue[j+1]-  queue[j]);
            seek += diff;
            System.out.println("Move " +queue[j] + " to " + queue[j+1] + " with seek " + diff);

        }

        System.out.println("Total Seek time is " + seek);
        avg = seek/(float)qSize;
        System.out.println("Average seek time is " +avg);

    }



}
